pkg/secrethub/dir.go

```diff
@@ -169,6 +174,35 @@ func (s dirService) Delete(path string) error {
 	return nil
 }
 
+// CreateAll creates all directories in the given path that do not exist yet.
+func (s dirService) CreateAll(path string) error {
+	err := api.ValidateDirPath(path)
+	if err != nil {
+		return err
+	}
+	return s.createAll(path)
+}
+
+func (s dirService) createAll(path string) error {
+	if len(strings.Split(path, "/")) < 3 {
+		return nil
+	}
+
+	_, err := s.GetTree(path, 0, false)
+	if err != api.ErrDirNotFound {
+		return err
+	}
+
+	_, err = s.Create(path)
+	if err == api.ErrDirAlreadyExists {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	return s.createAll(secretpath.Parent(path))
```

Comment:
Could you walk me through the logic here? For instance take the path `company/repo/foo/bar/baz`, then does it call `s.Create(path) in this order:

1. `s.Create("company/repo/foo/bar/baz")`
2. `s.Create("company/repo/foo/bar")`
1. `s.Create("company/repo/foo")`

If it does call it in that order, it seems to me like it would fail, correct?

Reply:
You're right! fixed it and added a test.